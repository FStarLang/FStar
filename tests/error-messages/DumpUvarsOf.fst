module DumpUvarsOf

open FStar.Tactics.V2

assume val p : bool -> int -> Type
assume val q : Type
assume val lem : b:bool -> i:int -> squash (p b i) -> Lemma q
assume val p5 : unit -> Lemma (p false 5)

let test () =
  assert q by begin
    apply_lemma (`lem);
    dump "1";
    dump_uvars_of (_cur_goal ()) "2";
    apply_lemma (`p5);
    ()
  end
