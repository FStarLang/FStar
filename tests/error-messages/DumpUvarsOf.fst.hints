[ "ÛCu?r¥ÝBÙ|×Ggóýw", [] ]