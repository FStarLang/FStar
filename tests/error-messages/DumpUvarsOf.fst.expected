proof-state: State dump @ depth 1 (1):
Location: DumpUvarsOf.fst(13,4-13,12)
Goal 1/1:
(_: Prims.unit) |- _ : Prims.squash (DumpUvarsOf.p (*?u89*) _ (*?u90*) _)

proof-state: State dump @ depth 1 (2):
Location: DumpUvarsOf.fst(14,4-14,36)
Goal 1/2:
(_: Prims.unit) |- _ : Prims.bool

Goal 2/2:
(_: Prims.unit) |- _ : Prims.int

Verified module: DumpUvarsOf
All verification conditions discharged successfully
